src/go/rpk/pkg/cli/version/version.go

```diff
@@ -11,27 +11,109 @@ package version
 
 import (
 	"fmt"
+	"runtime"
 
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
+	"go.uber.org/zap"
 )
 
+// Default build-time variables, passed via ldflags.
 var (
-	version string
-	rev     string
+	version   string // Semver of rpk.
+	rev       string // Short git SHA.
+	buildTime string // Timestamp of build time, ISO 8601.
+	hostOs    string // OS that was used to built rpk (go env GOHOSTOS).
+	hostArch  string // Arch that was used to built rpk (go env GOHOSTARCH).
 )
 
+type rpkVersion struct {
+	Version   string `json:"version,omitempty" yaml:"version,omitempty"`
+	GitRef    string `json:"git_ref,omitempty" yaml:"git_ref,omitempty"`
+	BuildTime string `json:"build_time,omitempty" yaml:"build_time,omitempty"`
+	GoVersion string `json:"go_version,omitempty" yaml:"go_version,omitempty"`
+	OsArch    string `json:"os_arch,omitempty" yaml:"os_arch,omitempty"`
+}
+
+type redpandaVersion struct {
+	NodeID  int
+	Version string
+}
+type redpandaVersions []redpandaVersion
+
 func Pretty() string {
 	return fmt.Sprintf("%s (rev %s)", version, rev)
 }
 
-func NewCommand() *cobra.Command {
-	command := &cobra.Command{
+func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
 		Use:   "version",
-		Short: "Check the current version",
-		Long:  "",
-		Run: func(_ *cobra.Command, _ []string) {
-			fmt.Println(Pretty())
+		Short: "Prints the current rpk and Redpanda version",
+		Long: `Prints the current rpk and Redpanda version.
+
+This command prints the current rpk version and allows you to list the Redpanda 
+version running on each node in your cluster.
+
+To list the Redpanda version of each node in your cluster you may pass the
+Admin API hosts via flags, profile, or environment variables.`,
+		Run: func(cmd *cobra.Command, _ []string) {
+			rv := rpkVersion{
+				Version:   version,
+				GitRef:    rev,
+				BuildTime: buildTime,
+				GoVersion: runtime.Version(),
+				OsArch:    fmt.Sprintf("%s/%s", hostOs, hostArch),
+			}
+			printRpkVersion(rv)
+			var rows redpandaVersions
+			defer printClusterVersions(&rows)
+
+			p, err := p.LoadVirtualProfile(fs)
+			if err != nil {
+				zap.L().Sugar().Errorf("unable to load the profile: %v", err)
+				return
+			}
+			cl, err := adminapi.NewClient(fs, p)
+			if err != nil {
+				zap.L().Sugar().Errorf("unable to create the admin client: %v", err)
+				return
+			}
+			bs, err := cl.Brokers(cmd.Context())
+			if err != nil {
+				zap.L().Sugar().Errorf("unable to request broker info: %v", err)
+				return
+			}
+			for _, b := range bs {
+				if b.IsAlive != nil {
+					rows = append(rows, redpandaVersion{b.NodeID, b.Version})
+				}
+			}
 		},
 	}
-	return command
+	return cmd
+}
+
+func printRpkVersion(rv rpkVersion) {
+	fmt.Printf(`rpk
```

Comment:
I like your idea in Slack to not lead with `rpk` and not have this section prefixed with two spaces